Build failure: Cutter

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

1. build cutter

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/326787764

### Relevant log output

```console
[ 77%] Building CXX object src/CMakeFiles/Cutter.dir/common/Decompiler.cpp.o
[ 78%] Building CXX object src/CMakeFiles/Cutter.dir/menus/AddressableItemContextMenu.cpp.o
/build/source/src/widgets/ColorThemeListView.cpp: In constructor 'ColorOptionDelegate::ColorOptionDelegate(QObject*)':
/build/source/src/widgets/ColorThemeListView.cpp:35:37: warning: 'void QGuiApplication::paletteChanged(const QPalette&)' is deprecated: Handle QEvent::ApplicationPaletteChange instead [-Wdeprecated-declarations]
   35 |     connect(qApp, &QGuiApplication::paletteChanged, this, [this]() {
      |                                     ^~~~~~~~~~~~~~
In file included from /nix/store/1q8sx67miwfn3ws5k7mkmkcjbym4akkp-qtbase-6.11.0/include/QtGui/qscreen_platform.h:19,
                 from /nix/store/1q8sx67miwfn3ws5k7mkmkcjbym4akkp-qtbase-6.11.0/include/QtGui/qscreen.h:150,
                 from /nix/store/1q8sx67miwfn3ws5k7mkmkcjbym4akkp-qtbase-6.11.0/include/QtGui/QScreen:1,
                 from /build/source/src/widgets/ColorThemeListView.cpp:8:
/nix/store/1q8sx67miwfn3ws5k7mkmkcjbym4akkp-qtbase-6.11.0/include/QtGui/qguiapplication.h:161:89: note: declared here
  161 |     QT_DEPRECATED_VERSION_X_6_0("Handle QEvent::ApplicationPaletteChange instead") void paletteChanged(const QPalette &pal);
      |                                                                                         ^~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/Cutter.dir/build.make:1927: src/CMakeFiles/Cutter.dir/plugins/PluginManager.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /build/source/src/widgets/GraphGridLayout.cpp:1:
In member function 'GraphGridLayout::GridBlock& GraphGridLayout::GridBlock::operator=(const GraphGridLayout::GridBlock&)',
    inlined from 'virtual void GraphGridLayout::CalculateLayout(GraphLayout::Graph&, long long unsigned int, int&, int&) const' at /build/source/src/widgets/GraphGridLayout.cpp:297:45:
/build/source/src/widgets/GraphGridLayout.h:37:12: warning: 'block' may be used uninitialized [-Wmaybe-uninitialized]
   37 |     struct GridBlock
      |            ^~~~~~~~~
/build/source/src/widgets/GraphGridLayout.cpp: In member function 'virtual void GraphGridLayout::CalculateLayout(GraphLayout::Graph&, long long unsigned int, int&, int&) const':
/build/source/src/widgets/GraphGridLayout.cpp:295:19: note: 'block' declared here
  295 |         GridBlock block;
      |                   ^~~~~
make[1]: *** [CMakeFiles/Makefile2:112: src/CMakeFiles/Cutter.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.22, NixOS, 26.05 (Yarara), 26.05.20260414.4bd9165`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.95.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches

### Notify maintainers



---
@Mic92 
**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/analysis/rizin/cutter.nix
+++ b/pkgs/development/tools/analysis/rizin/cutter.nix
@@ -1,7 +1,7 @@
 {
   lib,
   fetchFromGitHub,
-  fetchpatch,
+  fetchpatch2,
   stdenv,
   # for passthru.plugins
   pkgs,
@@ -34,6 +34,14 @@ let
       hash = "sha256-fNOznaFzWJ4Dve9U1+E4xPaznnyxae2jHNaBCdJzDyQ=";
       fetchSubmodules = true;
     };
+
+    patches = [
+      (fetchpatch2 {
+        name = "fix-shiboken6-type-index-case.patch";
+        url = "https://github.com/rizinorg/cutter/commit/07fea9c772dc573588dc2e5771f0740ee1883738.patch?full_index=1";
+        hash = "sha256-/C/s+Ui5F7MCxbzbChQ5Tv/oUHUQxXmk9xOnNI80xwQ=";
+      })
+    ];
 
     nativeBuildInputs = [
       cmake

